xarray_beam/_src/core.py

```diff
@@ -287,18 +301,31 @@ def __init__(
         rather than only on the host process. This is important for scaling
         pipelines to millions of tasks.
     """
+    if type(dataset) is xarray.Dataset:
+      dataset = (dataset,)
+    elif not dataset:
+      raise ValueError('dataset tuple cannot be empty!')
+    dataset = tuple(dataset)
+    if not _all_equal(ds.sizes for ds in dataset):
+      raise ValueError('all datasets must be the same size')
+    if split_vars and not _all_equal([(k, v.shape) for k, v in ds.items()]
+                                     for ds in dataset):
+      raise ValueError('when splitting variables, all datasets must have '
+                       'the same data variables with equivalent shapes')
     if chunks is None:
-      chunks = dataset.chunks
+      if not _all_equal(ds.chunks for ds in dataset):
+        raise ValueError('all datasets must have the same chunks or chunks must be provided')
+      chunks = dataset[0].chunks
```

Comment:
This validation logic is getting pretty long. I would suggest moving it to a helper function.